presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -88,6 +96,14 @@ public static double acos(@SqlType(StandardTypes.DOUBLE) double num)
         return Math.acos(num);
     }
 
+    @Description("arc cosine")
+    @ScalarFunction
+    @SqlType(StandardTypes.FLOAT)
+    public static long acos(@SqlType(StandardTypes.FLOAT) long num)
+    {
+        return Float.floatToRawIntBits((float) Math.acos(Float.intBitsToFloat((int) num)));
```

Comment:
Maybe call `MathFunctions.acos` instead for consistency?
I'd say if we don't have a native implementation for `float` but we cast it back and forth to call other `Math.function(double)` I think it's better to piggy-back on other existing Presto function and cast it into that function instead.